server/src/main/java/org/elasticsearch/rest/RestUtils.java

```diff
@@ -30,6 +31,12 @@
 
 public class RestUtils {
 
+    /**
+     * Sets whether we decode a '+' in an url as a space or not.
+     */
+    private static final boolean DECODE_PLUS_AS_SPACE =
+        Booleans.parseBoolean(System.getProperty("es.rest.url_plus_as_space", "true"));
```

Comment:
I think that we should immediately deprecate a true value here. Then in 8.0.0 we can reject true, allow for unset or false (with the latter deprecated), and then finally remove in 9.0.0 (let’s add assertions on the major version for this). Let us also add a note to the migration guide. That means after this PR is merged to master and backported to 7.x (only), we need a follow-up in master to reject false and do the deprecation and add the assertion.